modules/afsocket/afsocket-dest.c

```diff
@@ -234,6 +303,9 @@ afsocket_dd_connected(AFSocketDestDriver *self)
 
   proto = log_proto_client_factory_construct(self->proto_factory, transport, &self->writer_options.proto_options.super);
 
+  if (!_update_legacy_connection_persist_name(self, cfg->state))
```

Comment:
Can you move this out from `afsocket_dd_connected()`?
If the connection can't be established during the upgrade, we will lose the old entry (this code never runs and the entry will be deleted becaused it's unused).

Reply:
Thank You for pointing it out! I put it after `afsocket_dd_setup_transport` because we need to know if it is a stream or a dgram connection.